providers/fips/fipsindicator.c

```diff
@@ -16,17 +16,33 @@ void ossl_FIPS_INDICATOR_init(ossl_FIPS_INDICATOR *ind)
 {
     ind->strict_checks_set = 0;
     ind->strict_checks = 0;
-    ind->approved = -1;
+    ind->approved_set = 0;
+    ind->approved_flags = 0;
 }
 
-void ossl_FIPS_INDICATOR_set_approved(ossl_FIPS_INDICATOR *ind, int approved)
+void ossl_FIPS_INDICATOR_set_approved(ossl_FIPS_INDICATOR *ind, int id,
+                                      int approved)
 {
-    ind->approved = approved;
+    int mask = 1 << id;
+    ind->approved_set |= mask;
+    if (approved)
+        ind->approved_flags |= mask;
+    else
+        ind->approved_flags &= ~mask;
 }
 
-int ossl_FIPS_INDICATOR_get_approved(const ossl_FIPS_INDICATOR *ind)
+/* id_max is the highest index used */
+int ossl_FIPS_INDICATOR_get_approved(const ossl_FIPS_INDICATOR *ind, int id_max)
```

Comment:
Is the `id_max` argument necessary?
What happens if it isn't used?

The mask bit ops would change but I think they're possible: `return approved_set != 0 && ((approved_set + 1) & approved_set) == 0 ? 1 : -1`

Reply:
All approved flags are initially unknown. One or more may still be unknown when the allowed query happens...
And in that case we are still in the unknown state. So how do we know which ones are still unknown (considering all flags will be in this unknown state to start with).

Reply:
So the max was to try to get if we had any unknowns still. e.g. we use the bottom 3 bits (so these all have to be set before we can tell if the values are approved.